app/src/openxr/cpp/OpenXRInputSource.cpp

```diff
@@ -144,6 +147,33 @@ XrResult OpenXRInputSource::GetPoseState(XrAction action, XrSpace space, XrSpace
     return XR_SUCCESS;
 }
 
+XrResult OpenXRInputSource::applyHapticFeedback(XrAction action, XrDuration duration, float frequency, float amplitude) const
+{
+    XrHapticActionInfo hapticActionInfo { XR_TYPE_HAPTIC_ACTION_INFO };
+    hapticActionInfo.action = action;
+    hapticActionInfo.subactionPath = mSubactionPath;
+
+    XrHapticVibration hapticVibration { XR_TYPE_HAPTIC_VIBRATION };
+    hapticVibration.duration = duration;
+    hapticVibration.frequency = frequency;
+    hapticVibration.amplitude = amplitude;
+
+    RETURN_IF_XR_FAILED(xrApplyHapticFeedback(mSession, &hapticActionInfo, reinterpret_cast<const XrHapticBaseHeader *>(&hapticVibration)));
```

Comment:
Just do the type casting `(const XrHapticBaseHeader*)`. Also note that the `*` should be next to the type without any space.

Reply:
Fixed.